src/os/bluestore/BitAllocator.cc

```diff
@@ -724,24 +724,25 @@ void BitMapAreaIN::init(CephContext* const cct,
 
   m_child_size_blocks = level_factor;
 
-  BitMapArea **children = new BitMapArea*[num_child];
+  std::vector<BitMapArea*> children;
+  children.reserve(num_child);
   int i = 0;
   for (i = 0; i < num_child - 1; i++) {
     if (m_level <= 2) {
-      children[i] = new BitMapAreaLeaf(cct, m_child_size_blocks, i, def);
+      children.push_back(new BitMapAreaLeaf(cct, m_child_size_blocks, i, def));
     } else {
-      children[i] = new BitMapAreaIN(cct, m_child_size_blocks, i, def);
+      children.push_back(new BitMapAreaIN(cct, m_child_size_blocks, i, def));
     }
     total_blocks -= m_child_size_blocks;
   }
 
   int last_level = BitMapArea::get_level(cct, total_blocks);
   if (last_level == 1) {
-    children[i] = new BitMapAreaLeaf(cct, total_blocks, i, def);
+    children.push_back(new BitMapAreaLeaf(cct, total_blocks, i, def));
   } else {
-    children[i] = new BitMapAreaIN(cct, total_blocks, i, def);
+    children.push_back(new BitMapAreaIN(cct, total_blocks, i, def));
   }
-  BitMapAreaList *list = new BitMapAreaList(children, num_child);
+  BitMapAreaList *list = new BitMapAreaList(std::move(children));
```

Comment:
By 'typedefing' BitMapAreaList (see below) you might store items directly to this container instead of using interim children here and throughout..